libraries/standard/mqtt/src/mqtt.c

```diff
@@ -764,7 +1093,6 @@ static MQTTStatus_t sendPublish( MQTTContext_t * const pContext,
 
     assert( pContext != NULL );
     assert( pPublishInfo != NULL );
-    assert( headerSize > 0 );
```

Comment:
This is fine to keep here. Fix is in #979

Reply:
A rebase will handle it. For me to unblock my unit tests, I would need to have this change.

Reply:
If we rely on that, it's very likely that we'd forget about this, since deleting this line won't cause any merge conflicts in the rebase. If it's blocking you from running unit tests, build them with `cmake .. -DBUILD_TESTS="ON" -DCMAKE_C_FLAGS="-DNDEBUG"`